packages/@ember/-internals/metal/lib/mixin.ts

```diff
@@ -50,6 +51,36 @@ function isMethod(obj: any): boolean {
   );
 }
 
+function isAccessor(desc: PropertyDescriptor) {
+  return typeof desc.get === 'function' || typeof desc.set === 'function';
+}
+
+function extractAccessors(properties: { [key: string]: any } | undefined) {
+  if (properties !== undefined) {
+    let descriptors = getOwnPropertyDescriptors(properties);
+    let keys = Object.keys(descriptors);
+    let hasAccessors = keys.some(key => isAccessor(descriptors[key]));
```

Comment:
1. I don't think we can rely on `.some`
2. I'd prefer to avoid checking the same thing for `isAccessor` twice

Suggest:

```js
let accessors = keys.filter(key => isAccessor(descriptors[key]));

if (accessors.length > 0) {
  // ...snip...
}
```

What do you think?

Reply:
The reason why I run over the object twice is it's also part of the cloning process. We can't do `extracted = assign({}, properties)` because that'll actually trigger any native getters that exist there, we have to skip over native getters. This means we need to iterate the full list of properties.

However, we _don't_ want to start cloning eagerly, and create a new object, unless one of the values in the object is an accessor. So we need to check to see if anything is an accessor first, then we need to loop over all the keys again to assign them conditionally, based on whether or not one is an accessor.

We could alternatively mutate the `properties` object in place' but in order to do that we would need to use `Object.defineProperty` since we're talking about overriding native getters/setters:

```js
let accessors = keys.filter(key => isAccessor(descriptors[key]));

accessors.forEach(key => {
  Object.defineProperty(properties, key, { value: descriptors[key] });
})
```

I figured the double whammy of using `Object.defineProperty` (which I understand to be slow-ish) and mutating the object in place didn't make sense, but happy to change it if that makes more sense

Reply:
Gotcha, yes I agree. Also, I was surprised that IE9+ has `Array.prototype.some`.